### ISSUE TITLE
Selection sometimes moves randomly from selected email in thread list

### ISSUE THREAD

**Issue by jackiehluo** [nylas/N1#2825]
>Every now and then I hit this bug where if I select a message from a folder, 
the "selection" doesn't stick, and it travels up three or four messages. In 
effect, it's like the app things I'm holding down the UP arrow from my 
keyboard.

https://nylas.zendesk.com/agent/tickets/6147

**Comment by Ehekatl** [comment]
confirm this on windows 10 with recent version (currently I'm on 0.4.51), this last more then one month

**Comment by jffrybrwn** [comment]
This has been happening for me on Mac OS X El Capitan for the latest versions over the past month or so.

**Comment by dubstrike** [comment]
This has been happening from the start when selecting unread mail. It just travels a couple of messages up and marks them as read. **Extremely annoying.** 🔥 

- OS X 10.11.6
- 0.4.51-93ef779 (0.4.51-93ef779)

**Comment by smashhuevo** [comment]
I've been having this problem on macOS Sierra for the last couple of months.  Seems to happen if I bulk-select a bunch of items while syncing is going on.  I then hit "archive," and the wrong e-mails archive.

**Comment by dubstrike** [comment]
Any development on this? @jackiehluo
I keep experiencing this bug multiple times per day.

**Comment by jffrybrwn** [comment]
Or acknowledgment from the dev team?

Last night the selection scrolled beyond the top of the visible list.

If there's something we can do to help isolate this so you can fix it, please let us know!

**Comment by jackiehluo** [comment]
@dubstrike Are you in the Unread view?

**Comment by dubstrike** [comment]
@jackiehluo Yes, that's what I use mostly.

**Comment by jackiehluo** [comment]
@dubstrike Ahhh, so that specifically is a bug we haven't been able to track yet—#2271. Is anyone in this thread having this problem in another view?

**Comment by Artjaazz** [comment]
Confir the issue, same here, goes up two-three emails. Windows 10, Unread view

**Comment by laughedelic** [comment]
Same on OS X El Capitan, N1 v0.4.57.

**Comment by a-grealish** [comment]
I believe this is also affected by the "When reading message, mark as read: After X seconds." setting. If I have this set to 2 seconds it jumps up one message after 2 seconds. If the setting is "immediately" it jumps up very quickly through multiple messages.

I think this issue may also be the same as #2131. I seems correct that the scrolling happens until the loading of the thread is complete.

**Comment by dubstrike** [comment]
@a-grealish That's correct. It doesn't seem to be happening after I disable automatic marking as read.

**Comment by koenpunt** [comment]
I did some investigation, and found that this is happening because the message isn't available in the "unread" set after being read, for which in the thread list is accounted for by tracking recently read threads (`RecentlyReadStore`), but doesn't do this in the `ThreadListStore`.
I tried to fix this in [#3097](https://github.com/nylas/N1/pull/3097/commits/80188710678eef61e5214e036f6c98c3f39df17b), but apparently the `RecentlyReadStore` isn't updated before there.

**Comment by jyboudreau** [comment]
I'm seeing this issue almost constantly. It's making N1 pretty unusable. I've gone back to my other email client until this is fixed.

**Comment by eljenso** [comment]
Just discovered N1 but this issue makes it unusable for me as well.
Will switch back to my old client for now.

Probably doesn't matter but I'm running Fedora 25 with Gnome 3.22.

**Comment by inuwashi** [comment]
Again, confirm and poke.
Ubuntu Linux, Ver 1.5.0 pro, this is preventing you from being the best email client by a mile for this echo systems.

Also not random, always up the list.

**Comment by matteovega** [comment]
Poke and confirm for 1.5.0 on macOS 10.12.4 Beta. Only seems to happen in the Unread view.

**Comment by sethwebster** [comment]
+1 - very consistently happening.

Here's a video of it happening. This took 1 attempt.

https://www.dropbox.com/s/sw80jhff1cvsxhy/NylasN1SelectionTravel.mov?dl=0

**Comment by andrewalexander** [comment]
Adding that it is still doing this on Ubuntu 16.04. 

I noticed it only seems to do it when navigating to an unread message. Seems some sort of conflicting behavior after it's marked as read where it's instructing the selection to move somewhere (complete shot in the dark without even looking at the code).

**Comment by pugson** [comment]
@andrewalexander This project is dead. This bug has been fixed in https://github.com/Foundry376/Mailspring

**Comment by laughedelic** [comment]
> This bug has been fixed in Foundry376/Mailspring

@andrewalexander you should be aware that there is another fork: https://github.com/nylas-mail-lives/nylas-mail and I think that this problem was solved there too, but I'm not sure.
